mozilla_version/test/test_gecko.py

```diff
@@ -742,3 +742,1141 @@ def test_gecko_version_hashable():
     It is possible to hash `GeckoVersion`.
     """
     hash(GeckoVersion.parse('63.0'))
+
+
+@pytest.mark.parametrize('version_string', (
+    "1.0",
+    "1.5",
+    "2.0",
+    "3.0",
+    "3.5",
+    "3.6",
+    "4.0",
+    "13.0",
+    "14.0.1",
+    "14.0.1-build1",
+    "15.0",
+    "33.0",
+    "33.1",
+    "33.1-build1",
+    "33.1-build2",
+    "33.1-build3",
+    "34.0",
+    "124.0",
+    "125.0.1",
+    "125.0.1-build1",
+))
+def test_firefox_version_are_major(version_string):
+    assert FirefoxVersion.parse(version_string).is_major
+
+
+@pytest.mark.parametrize('version_string', (
+    "1.0",
+    "1.1",
+    "2.0",
+    "14.0.1",
+    "33.1",
+    "33.1-build1",
+    "68.1",
+    "68.1-build1",
+))
+def test_fennec_version_are_major(version_string):
+    assert FennecVersion.parse(version_string).is_major
+
+
+@pytest.mark.parametrize('version_string', (
+    "1.0",
+    "1.5",
+    "2.0",
+    "3.0",
+    "3.1",
+    "5.0",
+    "31.0",
+    "38.0.1",
+    "38.0.1-build1",
+    "45.0",
+    "115.0",
+))
+def test_thunderbird_version_are_major(version_string):
+    assert ThunderbirdVersion.parse(version_string).is_major
+
+
+@pytest.mark.parametrize('version_string', (
+    "1.0.1",
```

Comment:
Note to future readers: Unlike the version above where I singled out edge cases, here I just dumped what product-details had. It's not great but it saved time and I'm sure all existing cases didn't regress.